shell/platform/embedder/embedder.h

```diff
@@ -143,20 +143,39 @@ typedef struct {
   double pixel_ratio;
 } FlutterWindowMetricsEvent;
 
+// The kind of device generating a pointer event.
+typedef enum {
+  kMouse,
+  kSignal,
+} FlutterPointerDeviceKind;
```

Comment:
can't a signal come from a mouse?
does mouse mean trackpad also?

Reply:
There's no distinction between trackpad and mouse within the engine, so yes, mouse means trackpad. I'm not aware of a need to differentiate at this point.

As for kind, yes, this is overloading kind, under the assumption that the underlying device kind is irrelevant for pointer signals. (E.g., I don't know of a reason to distinguish between a mouse with a scroll wheel and a theoretical stylus with a scroll wheel).

There are lots of different ways we could approach this API. Some options:
- Make signal a phase instead. That's problematic because signals can themselves have phases (e.g., trackpad scroll has a down, moves, and up), so that would require a second signal_phase that could have the same values minus signal.
- Don't add signal to either phase or kind, but have a `none` value for `signal_kind` and use that to indicate whether it's a signal. I think that's cleaner than what I have now.
- Same, but add an `is_signal` field. Probably doesn't add anything over `none`.

I'll adjust to try the `none` option tomorrow and make sure there aren't any issues I'm missing. I think I just added the kind variant before realizing I needed a signal kind anyway, and hadn't revisited that.